cedar-lean/Cedar/Thm/Core/LT.lean

```diff
@@ -458,31 +462,31 @@ theorem ValueAttrs.lt_trans {vs₁ vs₂ vs₃: List (Attr × Value)}
   case inl.inl =>
     rcases h₁ with h₁ | h₁ <;> rcases h₂ with h₂ | h₂
     case inl.inl =>
-      rcases (String.strictLT.transitive a₁ a₂ a₃ h₁ h₂) with h₃
+      have h₃ := String.strictLT.transitive a₁ a₂ a₃ h₁ h₂
       simp [h₃]
     case inl.inr =>
-      rcases h₂ with ⟨h₂, _⟩ ; subst h₂ ; simp [h₁]
+      have ⟨h₂, _⟩ := h₂ ; subst h₂ ; simp [h₁]
     case inr.inl =>
-      rcases h₁ with ⟨h₁, _⟩ ; subst h₁ ; simp [h₂]
+      have ⟨h₁, _⟩ := h₁ ; subst h₁ ; simp [h₂]
     case inr.inr =>
-      rcases h₁ with ⟨hl₁, h₁⟩ ; subst hl₁
-      rcases h₂ with ⟨hl₂, h₂⟩ ; subst hl₂
-      rcases (Value.lt_trans h₁ h₂) with h₃
+      have ⟨hl₁, h₃⟩ := h₁ ; subst hl₁
+      have ⟨hl₂, h₄⟩ := h₂ ; subst hl₂
+      have h₃ := Value.lt_trans h₃ h₄
       simp [LT.lt] at h₃ ; simp [h₃]
   case inl.inr =>
-    rcases h₂ with ⟨h₂, _⟩
-    rcases h₂ with ⟨hl₂, hr₂⟩ ; subst hl₂ hr₂
+    have ⟨⟨hl₂, hr₂⟩, _⟩ := h₂
+    subst hl₂ hr₂
```

Comment:
here's an example of compound deconstruction with `have`.  (Replacing `rcases` code that wasn't even doing the compound deconstruction 😂 )